src/board/devices/output/7seg.ts

```diff
@@ -0,0 +1,205 @@
+  private isReg(): boolean {
+    return (
+      this.memory.readWord(this.startAddress).toUnsignedInteger() !=
+      this.oldSeg.toUnsignedInteger()
+    )
+  }
+  /**
+   * converts an number to an array of segment positions,true if seg on position should be on.
+   *
+   * @param inNum number to convert (0-15)
+   * @returns segmentArray an array of boolean, true if segment is turned on
+   */
+  private toSevensegNrBin(inNum: number): boolean[] {
+    let arr: boolean[] = []
+
+    let code: String = SEVENsegNr[inNum]
+    for (let i = 0; i < code.length; i++) {
+      const character = code.charAt(i)
+      if (character == '0') {
+        arr.push(false)
+      }
+      if (character == '1') {
+        arr.push(true)
+      }
+      if (character == '_') {
+      }
+    }
+
+    return arr
+  }
+
+  /**
+   * Returns an array of segment positions,true if seg on position should be on.
+   * the values are read from the datasection and interpreted binary
+   *
+   * @param display to check (0-3)
+   * @returns segmentArray an array of boolean, true if segment is turned on
+   */
+  private getBinDisplay(display: number): boolean[] {
+    let arr: boolean[] = []
+    switch (display) {
+      case 0:
+        arr = this.toSevensegNrBin(
+          this.memory.readByte(this.startAddressBin).toUnsignedInteger() % 16
+        )
+        this.oldBin = this.memory.readHalfword(this.startAddressBin)
+        break
+      case 1:
+        arr = this.toSevensegNrBin(
+          ~~(
+            this.memory.readByte(this.startAddressBin).toUnsignedInteger() / 16
+          )
+        )
+        break
+      case 2:
+        arr = this.toSevensegNrBin(
+          this.memory.readByte(this.endAddressBin).toUnsignedInteger() % 16
+        )
+        break
+      case 3:
+        arr = this.toSevensegNrBin(
+          ~~(this.memory.readByte(this.endAddressBin).toUnsignedInteger() / 16)
+        )
+        break
+      default:
+        //
+        break
+    }
+
+    return arr
+  }
+
+  /**
+   * converts an number to an array of segment positions,true if seg on position should be on.
+   *
+   * @param inByte byte to convert (0-F)
+   * @returns segmentArray an array of boolean, true if segment is turned on
+   */
+  private toSevensegNr(inByte: Byte): boolean[] {
+    let arr: boolean[] = []
+    let code: String = inByte.toBinaryString()
+    for (let i = 0; i < code.length; i++) {
+      const character = code.charAt(i)
+      if (character == '0') {
+        arr.push(false)
+      }
+      if (character == '1') {
+        arr.push(true)
+      }
+    }
+
+    return arr
+  }
+
+  /**
+   * Returns an array of segment positions,true if seg on position should be on.
+   * the values are read from the datasection and interpreted hexadecimal
+   *
+   * @param display to check (0-3)
+   * @returns segmentArray an array of boolean, true if segment is turned on
+   */
+  private getRegDisplay(display: number): boolean[] {
+    let arr: boolean[] = []
```

Comment:
And here I would just create the boolean array by making a for loop from 0..8 and check `isBitSet()` on the `Byte` that was read from the address of the display number.